### PR TITLE
Fixes SMES terminals and connection issues (PR by YoYoBatty)

### PR DESCRIPTION
**Pull request by ghost**
:cl: YoYoBatty
fix: SMES power terminals not actually deleting the terminal reference when by cutting the terminal itself rather than the SMES.
fix: SMES now reconnect to the grid properly after construction.
refactor: SMES now uses wirecutter act to handle terminal deconstruction.
/:cl:

https://github.com/HippieStation/HippieStation/pull/8238 i've been asked to push this upstream on behalf of yoyo
i don't really have much knowledge on the issue but i didn't see anything outright wrong in the code myself
tried doing cherry picking but couldn't get it working so i remade the PR entirely

### CODE REVIEW

**Review by AnturK — CHANGES_REQUESTED** [review]
The rest makes sense, but i don't understand this one.

File: code/modules/power/smes.dm [hunk 1]
```diff
@@ -160,6 +156,13 @@
 
 	return ..()
 
+/obj/machinery/power/smes/wirecutter_act(mob/living/user, obj/item/I)
+	//disassembling the terminal
+	if(terminal)
```
Comment (AnturK) [inline code/modules/power/smes.dm]: Also you removed the requirement of panel_open

Comment (McBawbaggings) [reply code/modules/power/smes.dm]: fixed

File: code/modules/power/smes.dm [hunk 2]
```diff
@@ -192,7 +195,6 @@
 /obj/machinery/power/smes/disconnect_terminal()
 	if(terminal)
 		terminal.master = null
-		terminal = null
```
[the remark this answers is not in the document]
Comment (McBawbaggings) [reply code/modules/power/smes.dm]: i assume yoyo thought he handled this in the dismantle proc
re-added

File: code/modules/power/terminal.dm [hunk 3]
```diff
@@ -71,6 +71,9 @@
 
 		new /obj/item/stack/cable_coil(drop_location(), 10)
 		to_chat(user, "<span class='notice'>You cut the cables and dismantle the power terminal.</span>")
+		var/obj/machinery/power/smes/S = master
+		if(istype(S, /obj/machinery/power/smes))
+			S.terminal = null
```
[the remark this answers is not in the document]
Comment (McBawbaggings) [reply code/modules/power/terminal.dm]: this is the terminal disconnecting it from the SMES when being dismantled and then deleting itself
i don't see what you believe to be the error in this part

[the remark this answers is not in the document]
Comment (AnturK) [reply code/modules/power/terminal.dm]: But it already calls disconnect terminal on destroy ?

[the remark this answers is not in the document]
Comment (McBawbaggings) [reply code/modules/power/terminal.dm]: you're right
i really shouldn't pr code that isn't mine even if asked
fixed

**Review by AnturK — APPROVED** [review]

**Comment by Dennok** [comment]
Sometimes after the dismantling of SMES there remains an indesmantlible terminal. This PR may fix it.

**Pull request merged**